spring-cloud-gcp-pubsub-stream-binder/src/test/java/org/springframework/cloud/gcp/stream/binder/pubsub/PubSubMessageChannelBinderTests.java

```diff
@@ -44,14 +44,15 @@
 
 	@BeforeClass
 	public static void enableTests() {
-		assumeTrue("PubSub Binder tests are disabled. Please enable them with -Dit.pubsub or -Dit.localdeps",
-				"true".equals(System.getProperty("it.pubsub")) || "true".equals(System.getProperty("it.localdeps")));
-		assumeThat(emulator.getEmulatorHostPort())
-				.withFailMessage("No PubSub emulator host configured").isNotNull();
+		assumeTrue("PubSub Binder tests are disabled. Please enable them with -Dit.pubsub or -Dit.emulator",
+				"true".equals(System.getProperty("it.pubsub")) || "true".equals(System.getProperty("it.emulator")));
 	}
 
 	@Override
 	protected PubSubTestBinder getBinder() {
+		if (emulator.getEmulatorHostPort() == null) {
+			fail("No PubSub emulator host found.");
```

Comment:
I think the assertion is not a responsibility of this method.
More over if we go false positive assumption in the `before()`,. we never step into this method.

Reply:
I agree that it's not a getter method's responsibility to check preconditions, but it's a convenient way to force all tests to fail when we either run on Travis CI or locally with a flag set.

In my mind, there are three scenarios:
1) test runs on Travis -- in this case, we want any environment issues to trigger test failure, breaking the build. Otherwise we can have tests skipping silently if, for example, emulator behavior changes. This behavior is achieved by having it.pubsub/it.emulator turned on by default on Travis.
2) running tests with no flags, and without cloud tools installed -- in this case, the binder tests should be skipped, since they cannot possibly run. This is your typical workflow, and it's addressed by the gating flags it.pubsub and it.emulator being off by default. Some of the emulator rule setup logic attempts to run, but only until the emulator rule realizes the environment is bad.
3) running tests locally with cloud tools installed -- this is my workflow, and I can just enable the it.emulator flag to achieve it.

So, not getting to the emulator check in getBinder() is intentional for scenario 2).